#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from backend.app.common.exception import errors
from backend.app.schemas.fuzz_test import *
from backend.app.database.db_mysql import async_db_session
from backend.app.crud.crud_case_group import CRUDCaseGroup
from backend.app.crud.crud_case import CRUDCase
from backend.app.crud.crud_primitive import CRUDPrimitive
from backend.app.schemas.fuzz_test import *

CaseGroupDao = CRUDCaseGroup()

class FuzzTestService:
    
    @staticmethod
    async def create_group(user_id, group_name, desc=None) -> None:
        async with async_db_session.begin() as db:
            if FuzzTestService.group_exists(db, user_id, group_name) is True:
                raise errors.ForbiddenError(msg='用例组已存在') 
            await CaseGroupDao.create_group(db, user_id, group_name, desc)
    
    @staticmethod
    async def group_exists(db: async_db_session, user_id, group_name) -> bool:

        group = await CaseGroupDao.get_group(db, user_id, group_name)
        return group is not None
            



